src/Raven.Studio/typescript/common/notifications/models/recentError.ts

```diff
@@ -35,21 +37,22 @@ class recentError extends abstractNotification {
         this.shortMessage = ko.pureComputed(() => generalUtils.trimMessage(this.message()));
     }
 
-    static tryExtractMessageAndException(details: string): { message: string, error: string } {
+    static tryExtractMessageAndException(details: string): { message: string, error: string, licenseLimitType: string } {
```

Comment:
This is quite unclean solution for me. I think it would be better to check if response status text is payment required and then call new method: extract licenseLimitDetails which will extract field message and licenselimit type from response json. Please remember this method is called not only for licensing errors.

Reply:
we're parsing the details anyway: JSON.parse(details)

Reply:
we do, but I think we should have different method for parsing and extracting license type. Both should be different return types.